src/safeposix/syscalls/net_calls.rs

```diff
@@ -755,73 +749,75 @@ impl Cage {
         }
     }
 
-    pub fn _cleanup_socket_inner(&self, filedesc: &mut FileDescriptor, partial: bool, shutdown: bool) -> i32 {
+    pub fn _cleanup_socket_inner(&self, filedesc: &mut FileDescriptor, how: i32, shutdown: bool) -> i32 {
+        let mut releaseflag = false;
         if let Socket(sockfdobj) = filedesc {
-            if let Some(localaddr) = sockfdobj.localaddr.as_ref().clone() {
-                let release_ret_val = NET_METADATA._release_localport(localaddr.addr(), localaddr.port(), sockfdobj.protocol, sockfdobj.domain);
-                sockfdobj.localaddr = None;
-                if let Err(e) = release_ret_val {return e;}
-                if !partial {
-                    if let Some(soid) = sockfdobj.socketobjectid {
-                        if shutdown {
-                            //we need to close the socket in order to send an EOF down it, but we
-                            //also need to have a valid socket object present and pointed to
-                            //otherwise table state gets corrupted/wonky
-                            let sockobjtherelock = NET_METADATA.socket_object_table.get(&soid).unwrap().clone();
-                            let mut sockobjthere = sockobjtherelock.write();
-
-                            //dropping the old socket closes it
-                            sockobjthere.0 = interface::Socket::new(sockfdobj.domain, sockfdobj.socktype, sockfdobj.protocol);
-                            sockobjthere.1 = ConnState::NOTCONNECTED;
-
-                            if let Some(localaddr) = sockfdobj.localaddr {
-                                if sockobjthere.0.bind(&localaddr) != 0 {
-                                    panic!("Bind on known ok address failed within shutdown!");
-                                }
-                            }
+            if let Some(soid) = sockfdobj.socketobjectid {
+                if shutdown {
 
-                            //check reuseaddr/port
-                            for optname in [sockfdobj.options & SO_REUSEPORT, sockfdobj.options & SO_REUSEADDR] {
-                                if optname != 0 {
-                                    if sockobjthere.0.setsockopt(SOL_SOCKET, optname, 1) < 0 {
-                                        panic!("Setsockopt within known ok conditions failed within shutdown!");
-                                    }
-                                }
-                            }
+                    let sockobjtherelock = NET_METADATA.socket_object_table.get(&soid).unwrap().clone();
```

Comment:
Is this really a lock? Small thing, but if I remember correctly the parking lot thing that I implemented isn't necessarily a lock. If that's true, maybe this should be named something else